src/bluesearch/database/download.py

```diff
@@ -213,6 +215,46 @@ def get_s3_urls(
     return url_dict
 
 
+def get_gcs_urls(
+    bucket: Bucket,
+    start_date: datetime,
+    end_date: datetime | None = None,
+) -> dict[str, HTTPIterator]:
+    """Get Google Cloud Storage urls.
+
+    Parameters
+    ----------
+    bucket
+        GCS bucket.
+    start_date
+        Starting date to download the incremental files (inclusive).
+    end_date
+        Ending date. If None, today is considered as the ending date (inclusive).
+
+    Returns
+    -------
+    url_dict
+        Keys represent different months. Values are iterators holding blobs
+        corresponding to actual PDF files.
+    """
+    date_list = get_daterange_list(
```

Comment:
As you wrote in the PR,  this `get_gcs_urls()` will fail if `start_date < datetime.datetime(2007, 04, 01)`.

So, what about adding a small check for that and, if the `start_date` is not ok then raise a `NotImplementedError` or `ValueError`?